repository-data/application/src/main/resources/hcm-config/configuration/update/Migration20180313UpdateGeographicCoverage.groovy

```diff
@@ -0,0 +1,108 @@
+package org.hippoecm.frontend.plugins.cms.admin.updater
+
+import org.onehippo.forge.content.exim.core.DocumentManager
+import org.onehippo.forge.content.exim.core.impl.WorkflowDocumentManagerImpl
+import org.onehippo.repository.update.BaseNodeUpdateVisitor
+import org.apache.jackrabbit.value.StringValue
+import javax.jcr.*
+
+class Migration20180313UpdateGeographicCoverage extends BaseNodeUpdateVisitor {
+
+    private static final String GEOGRAPHIC_COVERAGE_PROPERTY = "publicationsystem:GeographicCoverage"
+
+    List typesWithGeographicCoverage = [
+        "publicationsystem:publication",
+        "publicationsystem:dataset",
+        "publicationsystem:legacypublication",
+        "nationalindicatorlibrary:indicator"
+    ]
+
+    def geographicMap = ['England':['England'],
+                         'England and Northern Ireland':['England','Northern Ireland'],
+                         'England and Scotland':['England','Scotland'],
+                         'England and Wales':['England','Wales'],
+                         'England Scotland and Northern Ireland':['England','Scotland','Northern Ireland'],
+                         'England Wales and Northern Ireland':['England','Wales','Northern Ireland'],
+                         'Great Britain':['Great Britain'],
+                         'International':['International'],
+                         'Northern Ireland':['Northern Ireland'],
+                         'Scotland':['Scotland'],
+                         'UK':['UK'],
+                         'Wales':['Wales']
+    ]
+
+    boolean doUpdate(Node node) {
+
+        try {
+
+            // Just document nodes
+            if ("hippo:handle".equals(node.getPrimaryNodeType().getName())) {
+
+                if (node.hasNode(node.getName())) {
+                    Node subNode = node.getNode(node.getName());
+
+                    if (typesWithGeographicCoverage.contains(subNode.getPrimaryNodeType().getName())) {
```

Comment:
I wouldn't bother checking this here as you already check in the updateFieldAndSaveDocument method. Just call that method for all hippo:handles and let that decide if they have coverage to update or not